src/Parsing/Impl/Ast/GlobalStatement.cs

```diff
@@ -18,29 +18,30 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Python.Core;
+using Microsoft.Python.Core.Collections;
 
 namespace Microsoft.Python.Parsing.Ast {
     public class GlobalStatement : Statement {
-        private readonly NameExpression[] _names;
-
-        public GlobalStatement(NameExpression[] names) {
-            _names = names;
+        public GlobalStatement(ImmutableArray<NameExpression> names) {
+            Names = names;
         }
 
-        public IList<NameExpression> Names => _names;
+        public ImmutableArray<NameExpression> Names { get; }
+
+        public override IEnumerable<Node> GetChildNodes() => Names;
 
         public override void Walk(PythonWalker walker) {
             if (walker.Walk(this)) {
-                foreach (var n in _names.MaybeEnumerate()) {
-                    n?.Walk(walker);
+                foreach (var n in Names) {
+                    n.Walk(walker);
                 }
             }
             walker.PostWalk(this);
         }
 
         public override async Task WalkAsync(PythonWalkerAsync walker, CancellationToken cancellationToken = default) {
             if (await walker.WalkAsync(this, cancellationToken)) {
-                foreach (var n in _names.MaybeEnumerate().ExcludeDefault()) {
+                foreach (var n in Names) {
```

Comment:
I've seen names, asNames, etc being nulls a lot

Reply:
`asNames` can contain `null` for the case like `from a.b import c as cc, d, e as ee`. 
In this particular case, `GlobalStatement` is created only in `ParseGlobalStmt`, where `names` is created by `ReadNameList`, which fills array with `NameExpression` instances, produced by `MakeName`. So it can't have nulls.